src/erlfmt_scan.erl

```diff
@@ -180,6 +187,13 @@ last_node_string(#state{original = [First | _] = Tokens}) ->
     Location = erl_scan:location(First),
     {String, token_anno([{text, String}, {location, Location}])}.
 
+-spec last_node_string_trimmed(state()) -> {unicode:chardata(), anno()}.
```

Comment:
I wonder what happens when `last_node_string_trimmed` is used instead of `last_node_string` for all cases ?

Reply:
that's a good question. I don't fully understand the main read_nodes loop, so I'm not sure if trailing new lines can happen there and what affect of trimming them would have on formatting.

Reply:
In the interest of keeping the API small, what happens if you simply replace it everywhere, does anything break when you run `make test`?

Reply:
there are two cases failing, these are the diffs

```
$ diff output/no_pragma.erl test/erlfmt_SUITE_data/no_pragma.erl
1c1
< -module(no_pragma). % "שלום עולם" non-unicode trailing comment
\ No newline at end of file
---
> -module(no_pragma). % "שלום עולם" non-unicode trailing comment
```
```
$ diff output/unicode.erl test/erlfmt_SUITE_data/unicode.erl
1c1,3
< -module(unicode).%% we read ahead by one form, so we need this to test properly read_rest with what follows
---
> -module(unicode).
>
> %% we read ahead by one form, so we need this to test properly read_rest with what follows
```

Reply:
Interesting, these are both at the top, when we are still finding the pragma, so this might be fixable, but I think we can tackle that in a following diff.
I think this is ready to merge, but I'll wait for @michalmuskala to weigh in.